src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs

```diff
@@ -38,12 +38,15 @@ public async Task<IActionResult> CreateProvisionGap(CreateProvisionGapViewModel
         [Route("referral-create", Name = "CreateReferral")]
         public async Task<IActionResult> CreateReferral(string viewModel)
         {
-
             var createReferralViewModel = JsonConvert.DeserializeObject<CreateReferralViewModel>(viewModel);
 
             var dto = _mapper.Map<OpportunityDto>(createReferralViewModel);
-
-            var id = await _opportunityService.CreateOpportunity(dto);
+            
+            var id = createReferralViewModel.OpportunityId;
+            if (createReferralViewModel.OpportunityId > 0)
+                await _opportunityService.UpdateReferrals(dto);
```

Comment:
you also need to update opportunity because route, postcode and Distance might have changed

Reply:
If any of these are updated, then its classed as a new opportunity and opportunity id would be 0 and go down the CreateOpportunity path. I believe this is the functionality that Luke wants.